### DISCUSSION TITLE
Clear Table of All data

### DISCUSSION THREAD

**Discussion by conorw** [Budibase/budibase#4630] [Help] [answered]
Is there a quick way to clear all the data from an internal BB database table?

I have imported data from a CSV with errant phone numbers for all the records (they were missing a Zero from the start).
I now wish to reimport the data with the correct phone numbers. 
I tried to export the entire table from BB to a new CSV file, amend the numbers, then reimport, but this has just had the effect of creating duplicates for all the records.

Ideally, a mechanism for clearing the table of all data would be useful. Failing that, a way to multi-select all rows displayed onscreen in the data modeller would be helpful.

**Comment by mjashanks** [comment, marked as answer]
Hey Conor.

We don't have this functionality right now. Your options are to 
- Delete the table and re-import the data. If you have done a lot of Design work to your app, this would cause some bindings to break - But might not be too bad if you're only using the table in a few places. Export your app before you do this, as a backup!
- Multi-select records and delete. You can only do 10 at a time, due to paging :(
- Use our API to delete records. 

The API option will be a little tricky right now as we have not documented it at the moment (this is coming at the end of this month). But it's still possible to work it out by watching your dev tools / network tab, when you multi-delete a record.

**Comment by conorw** [comment]
Thanks @mjashanks, just checking before I start the manual process, is there any way to multiselect all items on the page e.g. with ctrl+a or some other konami code that will do it :))  
![image](https://user-images.githubusercontent.com/3536589/155311433-eab903f7-af7b-40c8-9ce3-d70f6daffc31.png)

**Reply by mjashanks** [reply]
nope, sorry!
